tsfresh/feature_extraction/extraction.py

```diff
@@ -313,16 +314,18 @@ def _do_extraction(df, column_id, column_value, column_kind,
                                     function_kwargs=kwargs)
     distributor.close()
 
-    # Return a dataframe in the typical form (id as index and feature names as columns)
-    result = pd.DataFrame(result)
-    if "value" in result.columns:
-        result["value"] = result["value"].astype(float)
+    return_df = pivot_list(result)
 
-    if len(result) != 0:
-        result = result.pivot("id", "variable", "value")
-        result.index = result.index.astype(df[column_id].dtype)
+    # All the features are floats
+    return_df = return_df.astype(float)
 
-    return result
+    # copy the type of the index
+    return_df.index = return_df.index.astype(df[column_id].dtype)
```

Comment:
for which type of index is that recasting necessary? I guess something like numbers that are saved as strings?

are we testing the types of indecies that need this recasting in some unit tests?

Reply:
Good question. If I remember correctly, that was mostly needed because of datetime columns.

I will add an issue to dig into this further, possible either removing this line or adding a unittest.
Thinking now a bit on it, I am not sure why this cast would trigger at all (because the type should already be the same).

Reply:
Raised in #706